pkg/plugins/plugin.go

```diff
@@ -154,6 +154,57 @@ func (p *Plugin) Wait() error {
 	return p.err
 }
 
+// AsyncReplyWithFunc is a wrapper function to provide a message handler function
+// which will asynchronously reply to the messages received by a plugin.
+// The difference between this and the synchronous ReplyWithFunc wrapper is that
+// while both receive messages in sequence, ReplyWithFunc will block on reading
+// the subsequent message from the inbox while the current message is being handled.
+func (p *Plugin) AsyncReplyWithFunc(fn SubscribeFunc, funcs ...filters.FilterFunc) error {
```

Comment:
What about an async reply function in the `context`. See here for reference https://github.com/andersnormal/autobot/blob/9169ac7307a5b7fcbce539cf20132988ed9821ab/pkg/plugins/context.go#L33. I think that would also be great if you could async reply in the context.